onnxruntime/contrib_ops/cpu/quantization/attention_quant.cc

```diff
@@ -91,10 +100,37 @@ Status QAttention<T>::PrePack(const Tensor& weights, int input_idx, bool& is_pac
     weights_data += head_size;
   }
 
+  bool kernel_owns_prepacked_buffer = (prepacked_weight_for_caching == nullptr);
+  if (!kernel_owns_prepacked_buffer) {
+    prepacked_weight_for_caching->buffers_.push_back(std::move(packed_weights_));
+    prepacked_weight_for_caching->buffer_sizes_.push_back(packed_weights_data_size);
+    prepacked_weight_for_caching->shapes_.push_back(weight_shape_);
+    prepacked_weight_for_caching->weights_sizes_.push_back(packed_weights_size_);
+    prepacked_weight_for_caching->is_filled_ = true;
+  }
+
   is_packed = true;
   return Status::OK();
 }
 
+template <typename T>
+Status QAttention<T>::StorePrePackedWeight(const PrepackedWeight& prepacked_weight,
+                                           int input_idx,
+                                           /*out*/ bool& stored_weight) {
+  stored_weight = false;
+
+  if (1 != input_idx) {
+    return Status::OK();
+  }
+
+  stored_weight = true;
+  weight_shape_ = prepacked_weight.shapes_[0];
+  packed_weights_size_ = prepacked_weight.weights_sizes_[0];
+  packed_weights_ = BufferUniquePtr(prepacked_weight.buffers_[0].get(), BufferDeleter(nullptr));
+
```

Comment:
Would it be clearer to have two members so the ownership semantics aren't slightly hidden by having a unique_ptr with a null deleter?

e.g.

const PrepackedWeight* prepacked_weights_;
std::unique_ptr<void> owned_prepacked_weights_;

Reply:
Not following this sorry. Are prepacked_weights_ and owned_prepacked_weights_ kernel members ?

Reply:
Fixed based on offline discussion